.claude/scripts/polarion_client.py

```
@@ -0,0 +1,407 @@
+            )
+
+            if response.status_code == 200:
+                # Parse SOAP response to extract test steps
+                # Use defusedxml to prevent XXE attacks
+                try:
+                    import defusedxml.ElementTree as ET
+                except ImportError:
+                    # Fallback to standard library with safe parsing
+                    import xml.etree.ElementTree as ET
+                    # Disable entity expansion to prevent XXE
+                    ET.XMLParser(resolve_entities=False)
```

Ruff (0.15.17)

[error] Using `xml` to parse untrusted data is known to be vulnerable to XML attacks; use `defusedxml` equivalents

(S314)